src/core/libmaven/csvreports.cpp

```diff
@@ -1,283 +1,150 @@
-                << "compoundId" << "formula" << "expectedRtDiff" << "ppmDiff" << "parent";
-        int cohort_offset = groupReportcolnames.size() - 1;
-        QString header = groupReportcolnames.join(SEP.c_str());
-        groupReport << header.toStdString();
-        for (unsigned int i = 0; i < samples.size(); i++) {
-            groupReport << SEP << sanitizeString(samples[i]->sampleName.c_str()).toStdString();
-        }
-        groupReport << endl;
-        //TODO: Remove this to remove row in csv reports --@Giridhari
-        if (includeSetNamesLine){
-             for(unsigned int i = 0; i < cohort_offset; i++) { groupReport << SEP; }
-             for(unsigned int i = 0; i < samples.size(); i++) { groupReport << SEP << sanitizeString(samples[i]->getSetName().c_str()).toStdString(); }
-             groupReport << endl;
-         }
+    if(groups.size() == 0){
+        errorReport = "No groups found!";
+        return 0;
     }
-    else {
-        errorReport = "Unable to write to file \"" + QString::fromStdString(outputfile) + "\"\n";
-        errorReport += "Please check if you have permission to write to the specified location or the file is not in use";
+    if(_fileName.length() < 4){
+        errorReport="invalid file name";
+        return 0;
     }
-}
-
-void CSVReports::insertPeakReportColumnNamesintoCSVFile(){
-
-    if (peakReport.is_open()) {
-        QStringList peakReportcolnames;
-        peakReportcolnames << "groupId" << "compound" << "compoundId" << "formula" << "sample" << "peakMz"
-                << "medianMz" << "baseMz" << "rt" << "rtmin" << "rtmax" << "quality"
-                << "peakIntensity" << "peakArea" << "peakSplineArea" << "peakAreaTop"
-                << "peakAreaCorrected" << "peakAreaTopCorrected"
-                << "noNoiseObs" << "signalBaseLineRatio"
-                << "fromBlankSample";
-        QString header = peakReportcolnames.join(SEP.c_str());
-        peakReport << header.toStdString() << endl;
+    string fileFormat = _fileName.substr(_fileName.length()-4 , 4 );
+    if( fileFormat == ".csv" )
+        SEP = ",";
+    if( fileFormat == ".tab" )
+        SEP = "\t";
+    outFileStream.open(_fileName);
+    if(!outFileStream.is_open()){
+        errorReport="file can't open to write\ncheck write permission";
+        return 0;
     }
-}
-
-//Feng note: CSVReports::addGroup modified to (1) output only C12 data without labeling or wen compound is unknown, (2) output all related isotopic forms with labeling,
-//even when peak height is zero
-
-void CSVReports::addGroup (PeakGroup* group) {
-
-      insertPeakInformationIntoCSVFile(group);
-
-      if(!groupReport.is_open()) {
-		return;
-	  }
-      //get ionization mode
-      insertGroupInformationIntoCSVFile(group);
-
-}
-
-void CSVReports::insertPeakInformationIntoCSVFile(PeakGroup* group) {
-
-      writePeakInfo(group);
+    else{
+        addColumnNames();
+        for(int i=0 ; i<groups.size(); ++i){
```

Comment:
Let's start using [range based for loop](http://en.cppreference.com/w/cpp/language/range-for) 

Same for every other for loop in CsvReports. We can ignore the rest of the classes for now

Reply:
@sahil21 @shubhra-agrawal what's your opinion here- shifting our traditional **for loop** to recommended by @rish9511

Reply:
@IpankajI It's not necessary but it would make for cleaner code with fewer iteration variables. Your call, I guess.